YtFlowApp/ConfigEncoding/ConfigEncoder.cs

```diff
@@ -31,5 +33,10 @@ private static string ShadowsocksConfigEncoder(ShadowsocksConfig config)
             }
             return $"ss://{url}{tag}";
         }
+
+        private static string TrojanConfigEncoder(TrojanConfig config)
+        {
+            return Uri.EscapeUriString($"trojan://{config.Password}@{config.ServerHost}:{config.ServerPort}");
```

Comment:
`Password` is the only field we have to escape, not the whole URL.

Reply:
But uri escape is right here.We can't assume that other parts will not contain characters that need to be encoded.`Uri.EscapeUriString` will good work for it.

Reply:
If the `password` contains an `@`, then it will break the parser. See https://github.com/trojan-gfw/trojan-url/blob/master/trojan-url.py#L85 .

Regarding the validation of other fields, we will hopefully do this when saving a configuration and assume the fields are valid.

Reply:
You are right.Fix it right now.